docs/0000-materialized-links.md

```diff
@@ -0,0 +1,187 @@
+---
+Start Date: 2020-07-31
+RFC PR: (leave this empty)
+Foam Issue: (leave this empty)
+---
+
+# 0002 Materialized Links
+
+## Summary
+
+In Foam, you can link between documents using `[[wiki-link]]` syntax. We parse all the markdown files (notes) in your Foam workspace, and create a graph data structure (not to be confused with the graph visualization) that links all the documents together. Each note has two kinds of links:
+
+- Forward links (links from that note to other notes)
+- Backlinks (links from other notes to that note)
+
+"Materialized links" is a Foam term for outputting the links from and to a document into the document itself. For benefits and use cases, see [Motivation](#motivation).
+
+## Basic example
+
+Consider following three markdown documents:
+
+`note-a.md`
+```md
+# Note A
+This is a link to [[note-b]].
+```
+
+`note-b.md`
+```md
+# Note B
+This links back to [[note-a]].
+```
+
+`note-c.md`
+```md
+# Note C
+This links to [[note-a]].
+```
+
+If we generated both Forward links and Backlinks for `note-a.md`, it would look like follows:
+
+```md
+# Note A
+This is a link to [[note-b]].
+
+## Foam:Links
+
+- [Note B](note-b)
+
+## Foam:Backlinks
+
+- [Note B](note-b)
+- [Note C](note-c)
+```
+
+## Motivation
+
+### Backlinks
+
+You can already view backlinks inside the Markdown Notes backlinks explorer. Materializing backlinks adds the following benefits:
+
+- Including the backlink information for benefit of other tools that do not have access to the Foam graph.
+- Used to generate reference lists
+  - E.g. "todo.md" can show every place `[[todo]]` is mentioned
+- Useful for publishing static websites, PDFs etc from your workspace and including backlink information.
+  - Note that a future [Foam static site generator](https://github.com/foambubble/rfcs/pull/2) may generate alternative representations of the backlinks, in which case materialization is not required for publishing.
+
+### Forward links
+
+There is currently no easy way to see which other notes a note links to.
+
+- This feature is useful for digital gardeners, who may want to include a link list at the end of their documents.
+- On its own, this feature may not be worth implementing, but as part of "materialized links", it makes sense to support linking both ways.
+
+## Detailed design
+
+### Link format
+
+Links are formatted as traditional Markdown links `[Title](target.md)`
+
+- This prevents a materialized backlink being treated as a regular link, which would cause every linking relationship automatically becoming two-way
```

Comment:
not sure I understand, does this means that it's not allowed for someone to use the `[Title](path/to/doc.md)` syntax for local files?